src/include/duckdb/main/materialized_query_result.hpp

```diff
@@ -24,6 +24,8 @@ class MaterializedQueryResult : public QueryResult {
 	DUCKDB_API MaterializedQueryResult(string error);
 
 	ChunkCollection collection;
+	//! In case we are streaming the data we keep track of the current chunk
+	idx_t cur_chunk = 0;
```

Comment:
The materialized chunk also supports `Fetch`, can't you just call that method for both the streaming and the materialized result?